core/src/main/java/com/opensymphony/xwork2/util/AnnotationUtils.java

```diff
@@ -139,19 +140,45 @@ public static String resolvePropertyName(Method method) {
      */
     public static <T extends Annotation> T findAnnotation(Class<?> clazz, Class<T> annotationClass) {
         T ann = clazz.getAnnotation(annotationClass);
-        while (ann == null && clazz != null) {
+        if (ann == null && clazz != Object.class) {
             ann = clazz.getAnnotation(annotationClass);
```

Comment:
As you replaced `while` with `if` then I think you can correct this method as below:

```java
    public static <T extends Annotation> T findAnnotation(Class<?> clazz, Class<T> annotationClass) {
        T ann = null;
        if (clazz != null) {
            ann = clazz.getAnnotation(annotationClass);
            if (ann == null) {
                ann = clazz.getPackage().getAnnotation(annotationClass);
            }
            if (ann == null) {
                ann = findAnnotation(clazz.getSuperclass(), annotationClass);
            }
        }

        return ann;
    }
```